lib/os_io.c

```diff
@@ -37,6 +37,199 @@ ___io_module ___io_mod =
 
 /*---------------------------------------------------------------------------*/
 
+/* poll dynamic fdset memory management. */
+
+#ifdef USE_select_or_poll
+
+___HIDDEN int ___fdset_heap_overflow;
+
+___HIDDEN int ___fdset_init
+   ___P((___processor_state ps),
+        (ps)
+        ___processor_state ps;)
+{
+  void *readfds = NULL;
+  void *writefds = NULL;
+  void *exceptfds = NULL;
+
+  int size = MAX_CONDVARS;
+  int bytes;
+
+  if (size < 8 * sizeof (fd_set))
+    size = 8 * sizeof (fd_set);
+  bytes = ___CEILING_DIV (size, 8);
+  
+  readfds  = ___ALLOC_MEM (bytes);
+  if (!readfds)
+    goto error;
+
+  writefds = ___ALLOC_MEM (bytes);
+  if (!writefds)
+    goto error;
+
+#ifdef USE_select
+  exceptfds = ___ALLOC_MEM (bytes);
+  if (!exceptfds)
+    goto error;
+#endif
+
+  memset (readfds, 0, bytes);
```

Comment:
Yes, they are zeroed out by `___device_select`, so strictly speaking it's unnecessary.
I just like to have my pointers clean for debugging purposes.
I can certainly remove the memset if you think it's unnecessary.